jujugui/static/gui/src/app/components/search-results-item/search-results-item.js

```diff
@@ -0,0 +1,238 @@
+      );
+    },
+
+    /**
+      Generate the elements for the icon list.
+
+      @method _generateIconList
+      @returns {String} The generated elements.
+    */
+    _generateIconList: function() {
+      var services = this.props.item.services || [];
+      var components = [];
+      if (services.length === 0) {
+        services.push(this.props.item);
+      }
+      services.forEach(function(service) {
+        var src = service.iconPath ||
+            'juju-ui/assets/images/non-sprites/charm_160.svg';
+        components.push(
+          <li className="list-icons__item"
+            key={service.displayName}
+            role="button" tabIndex="0"
+            title={service.displayName}
+            onClick={this._handleItemClick.bind(this, service.id)}>
+            <img src={src}
+              className="list-icons__image"
+              alt={service.displayName} />
+          </li>
+        );
+      }, this);
+      return components;
+    },
+
+    /**
+      Generate the base classes from the props.
+
+      @method _generateClasses
+      @param {Boolean} selected Whether the filter is selected.
+      @returns {String} The collection of class names.
+    */
+    _generateClasses: function(selected) {
+      return classNames(
+        {selected: selected}
+      );
+    },
+
+    /**
+      Show the entity details when clicked.
+
+      @method _handleEntityClick
+      @param {String} id The entity id.
+      @param {Object} e The click event.
+    */
+    _handleItemClick: function(id, e) {
+      e.stopPropagation();
+      this.props.changeState({
+        sectionC: {
+          component: 'charmbrowser',
+          metadata: {
+            activeComponent: 'entity-details',
+            id: id
+          }
+        }
+      });
+    },
+
+    /**
+      Filter by the given tag.
+
+      @method _handleTagClick
+      @param {String} tag The tag name.
+      @param {Object} e The click event.
+    */
+    _handleTagClick: function(tag, e) {
+      e.stopPropagation();
+      this.props.changeState({
+        sectionC: {
+          component: 'charmbrowser',
+          metadata: {
+            activeComponent: 'search-results',
+            tags: tag
+          }
+        }
+      });
+    },
+
+    /**
+      Filter by the given series.
+
+      @method _handleSeriesClick
+      @param {String} series The series name.
+      @param {Object} e The click event.
+    */
+    _handleSeriesClick: function(series, e) {
+      e.stopPropagation();
+      this.props.changeState({
+        sectionC: {
+          component: 'charmbrowser',
+          metadata: {
+            activeComponent: 'search-results',
+            series: series
+          }
+        }
+      });
+    },
+
+    render: function() {
+      var item = this.props.item;
+      return (
+        <li className={'list-block__list--item ' + item.type}
+            tabIndex="0" role="button"
+            onClick={this._handleItemClick.bind(this, item.storeId)}>
```

Comment:
React complains about .bind() could this instead just reference `this.props.item.storeId` in `_handleItemClick`?